src/Microsoft.OData.Core/UriParser/ExpressionLexer.cs

```diff
@@ -447,6 +453,24 @@ internal string AdvanceThroughBalancedParentheticalExpression()
             return expressionText;
         }
 
+        internal void TakeTextPositionSnapshot()

```

Comment:
Should we change the behavior of the method too?
I mean, what happens if we return false? We override the current snapshot or leave it as it is?